src/main/java/com/synopsys/integration/alert/web/user/RoleActions.java

```diff
@@ -84,7 +84,9 @@ public UserRoleModel createRole(RolePermissionModel rolePermissionModel) throws
         Set<PermissionModel> permissions = rolePermissionModel.getPermissions();
         validatePermissions(permissions);
         PermissionMatrixModel permissionMatrixModel = convertToPermissionMatrixModel(permissions);
-        return authorizationUtility.createRoleWithPermissions(roleName, permissionMatrixModel);
+        UserRoleModel userRoleModel = authorizationUtility.createRoleWithPermissions(roleName, permissionMatrixModel);
+        authorizationManager.loadPermissionsIntoCache();
```

Comment:
This is a bug that might need to be patched in older versions.  The cache doesn't get updated when a new role is created.  Therefore this descriptor list for the user in the UI would be empty causing the original issue.